src/setupEventStream.ts

```diff
@@ -0,0 +1,86 @@
+import { InternalException } from './utils/InternalException';
+import { systemError } from './utils/Logger';
+import { nonNullProp } from './utils/nonNull';
+import { WorkerChannel } from './WorkerChannel';
+import LogCategory = rpc.RpcLog.RpcLogCategory;
+import LogLevel = rpc.RpcLog.Level;
+
+/**
+ * Configures handlers for incoming gRPC messages on the client
+ *
+ * This should have a way to handle all incoming gRPC messages.
+ * This includes all incoming StreamingMessage types (exclude *Response types and RpcLog type)
+ */
+export function setupEventStream(
+    workerId: string,
+    eventStream: IEventStream,
+    functionLoader: IFunctionLoader
+): WorkerChannel {
+    const channel = new WorkerChannel(eventStream, functionLoader);
+
+    eventStream.on('data', (msg) => {
+        const eventName = msg.content;
+        switch (eventName) {
+            case 'functionEnvironmentReloadRequest':
+                functionEnvironmentReloadRequest(channel, msg.requestId, nonNullProp(msg, eventName));
+                break;
+            case 'functionLoadRequest':
+                void functionLoadRequest(channel, msg.requestId, nonNullProp(msg, eventName));
+                break;
+            case 'invocationRequest':
+                invocationRequest(channel, msg.requestId, nonNullProp(msg, eventName));
+                break;
+            case 'workerInitRequest':
+                workerInitRequest(channel, msg.requestId, nonNullProp(msg, eventName));
+                break;
+            case 'workerStatusRequest':
+                workerStatusRequest(channel, msg.requestId, nonNullProp(msg, eventName));
+                break;
+            case 'closeSharedMemoryResourcesRequest':
+            case 'fileChangeEventRequest':
+            case 'functionLoadRequestCollection':
+            case 'functionsMetadataRequest':
+            case 'invocationCancel':
+            case 'startStream':
+            case 'workerHeartbeat':
+            case 'workerTerminate':
+                // Not yet implemented
+                break;
+            default:
+                channel.log({
+                    message: `Worker ${workerId} had no handler for message '${eventName}'`,
+                    level: LogLevel.Error,
+                    logCategory: LogCategory.System,
+                });
+        }
+    });
+
+    eventStream.on('error', function (err) {
+        systemError(`Worker ${workerId} encountered event stream error: `, err);
+        throw new InternalException(err);
+    });
+
+    // wrap event stream write to validate message correctness
+    const oldWrite = eventStream.write;
+    eventStream.write = function checkWrite(msg) {
+        const msgError = rpc.StreamingMessage.verify(msg);
+        if (msgError) {
+            systemError(`Worker ${workerId} malformed message`, msgError);
+            throw new InternalException(msgError);
+        }
+        oldWrite.apply(eventStream, [msg]);
+    };
+
+    return channel;

```

Comment:
Do we need this return value? If we do, I would prefer delegating the creation of the channel to `WorkerChannel.ts` then passing it to this function as an argument. `setupEventStream` for me doesn't imply that it's also creating the worker channel on my behalf and returning it to me

Reply:
Addressed this

It's probably worth noting that I expect more refactoring work related to `WorkerChannel` moving forward. For example, I've removed so much from it that I'm not really sure it's a "Channel" anymore. One idea I have is to rename it to `WorkerContext`, basically the same thing as `InvocationContext` just at the worker scope. But that would be a separate PR